packages/payment-integration-test-framework/src/fixture/pagePreset/ApiRequestsSender.ts

```diff
@@ -103,55 +117,76 @@ export class ApiRequestsSender {
-        const consignmentId = cartWithConsignments.consignments[0].id as string;
+        const consignmentId = cartWithConsignments.consignments[0].id;
         const shippingOptionId =
             cartWithConsignments.consignments[0].availableShippingOptions[0].id;
 
+        if (!shippingOptionId) {
+            throw new Error(
+                `Unable to select a shipping option for the address: ${JSON.stringify(
+                    address,
+                )}.\nPlease check shipping configuration.`,
+            );
+        }
+
         await apiContext.put(`./checkouts/${checkout.id}/consignments/${consignmentId}`, {
             params: { include: 'consignments.availableShippingOptions' },
             data: { shippingOptionId },
         });
 
         // Set billing address
-        const billingAddressId = checkout.billingAddress?.id || '';
-
-        await apiContext.put(`./checkouts/${checkout.id}/billing-address/${billingAddressId}`, {
-            data: {
-                firstName: faker.name.firstName(),
-                lastName: faker.name.lastName(),
-                email: faker.internet.email('checkout'),
-                company: faker.company.companyName(),
-                address1: faker.address.streetName(),
-                address2: faker.address.secondaryAddress(),
-                city: faker.address.cityName(),
-                stateOrProvinceCode: stateCode,
-                countryCode: 'US',
-                postalCode: faker.address.zipCodeByState(stateCode),
+        console.log(`  - Setting random billing address`);
+
+        const billingAddressId = checkout.billingAddress?.id;
+
+        await apiContext.put(
+            `./checkouts/${checkout.id}/billing-address/${billingAddressId ?? ''}`,
+            {
+                data: {
+                    firstName: faker.name.firstName(),
+                    lastName: faker.name.lastName(),
+                    email: faker.internet.email('checkout'),
+                    company: faker.company.companyName(),
+                    address1: faker.address.streetName(),
+                    address2: faker.address.secondaryAddress(),
+                    phone: faker.phone.phoneNumber('##########'),
+                    city: faker.address.cityName(),
+                    stateOrProvinceCode: stateCode,
+                    countryCode: 'US',
+                    postalCode: faker.address.zipCodeByState(stateCode),
+                },
             },
-        });
+        );
     }
 
-    async dispose(): Promise<void> {
+    async dispose(message: string): Promise<void> {
+        console.log(`Disposing APIRequestContext`);
+
         const apiContext = await this.apiContextFactory.create(this.page, this.storeUrl);
 
         await apiContext.dispose();
+
+        const time = Math.ceil((Date.now() - this.startTime) / 1000);
+
+        console.log(`\x1b[32m\n  ${message} \x1b[0m\x1b[2m(${time}s)\x1b[0m`);
     }
```

Comment:
Updated look of recording Bigpay testing payment method.
<img width="500" alt="image" src="https://user-images.githubusercontent.com/88361607/198202098-ee18079e-fddf-4ce3-af36-4368baacef1a.png">